felix/fv/iptables_force_programming_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2023 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build fvtests
+
+package fv_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	api "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+
+	"github.com/projectcalico/calico/felix/fv/connectivity"
+	"github.com/projectcalico/calico/felix/fv/infrastructure"
+	"github.com/projectcalico/calico/felix/fv/utils"
+	"github.com/projectcalico/calico/felix/fv/workload"
+	"github.com/projectcalico/calico/libcalico-go/lib/apiconfig"
+	client "github.com/projectcalico/calico/libcalico-go/lib/clientv3"
+)
+
+var _ = infrastructure.DatastoreDescribe("iptables force-programming tests", []apiconfig.DatastoreType{apiconfig.Kubernetes}, func(getInfra infrastructure.InfraFactory) {
+	var (
+		tc     infrastructure.TopologyContainers
+		infra  infrastructure.DatastoreInfra
+		client client.Interface
+		w      [3]*workload.Workload
+		cc     *connectivity.Checker
+	)
+
+	BeforeEach(func() {
+		if NFTMode() {
+			Skip("This test is not yet supported in NFT mode")
```

Comment:
I wondered if the hint might cause problems for NFT mode.  Do we guard higher up against passing the hint down in NFT mode, e.g. from the calc graph?

Reply:
(In the non-test Felix code, I mean.)

Reply:
No, it simply won't touch that field AFAICT.  Might make sense to implement for nft too but I thoght I'd start with iptables.